libcob/common.c

```diff
@@ -7817,12 +7858,58 @@ cob_expand_env_string (char *strval)
 				k++;
 			}
 			k--;
-		} else if (strval[k] == '$'
-		        && strval[k+1] == '$') {	/* Replace $$ with process-id */
-			j += sprintf (&env[j], "%d", cob_sys_getpid());
-			k++;
-		/* CHECME: possibly add $f /$b as basename of executable [or, when passed to cob_init the first name] 
-		           along with $d date as yyyymmdd and $t as hhmmss */
+		} else if (strval[k] == '$') {
+			struct cob_time time;
+			char *need_free = NULL;
+			const char *s = NULL;
+		        switch ( strval[k+1] ){
+			case '$': /* Replace $$ with process-id */
+				j += sprintf (&env[j], "%d", cob_sys_getpid());
+				k++;
+				break;
+			case 'f': /* $f is the executable filename */
+				if (!cobglobptr->cob_main_argv0){
+					env[j++] = strval[k];
+				} else {
+					s = cobglobptr->cob_main_argv0;
+				}
+				break;
```

Comment:
Problem is that `$f` is an absolute file path and `$$` is the pid ; either I have to print dummy data (when `is_test` is true) or post-process the output in the `.at` file...

Reply:
dummy data for $$ like 123456 (for all cases) seems reasonable; for $f you may check if `cobcrun -q` provides an option (falling back to base name in `cobglobptr->cob_main_argv0` as well)

Reply:
While `cobcrun -q` does set `argv[0]` to a base name, `cob_set_main_argv0` does not always use that information (eg. when using readlink). I could change `cob_set_main_argv0` to return only a base name when running in test mode, but I think this could reduce the coverage of this function.

Reply:
Thank you for checking, I agree to not do that adjustment.

For testing `cob_set_main_argv0` - what do you think of a profile run with that included in the name, then doing an `ls 2>&1 1>/dev/null` with the basename, checking the return code for zero [this could be done before the test that now includes the pseudo-pid 123456 without checking the file content, as that will be done in the next `AT_CHECK` (the one that already exists)]?

Reply:
That seems to work - I'll do that.